plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -1909,6 +1911,209 @@ read_only::get_table_rows_result read_only::get_table_rows( const read_only::get
+
+   // Check valid table name
+   const auto table_it = abi.kv_tables.value.find(p.table);
+   if( table_it == abi.kv_tables.value.end() ) {
+     EOS_ASSERT(false, chain::contract_table_query_exception,  "Unknown kv_table: ${t}", ("t", tbl_name));
+   }
+   const auto &kv_tbl_def = table_it->second;
+   // Check valid index_name
+   bool is_primary_idx = (p.index_name == kv_tbl_def.primary_index.name);
+   bool is_sec_idx = (kv_tbl_def.secondary_indices.find(p.index_name) != kv_tbl_def.secondary_indices.end());
+   EOS_ASSERT(is_primary_idx || is_sec_idx, chain::contract_table_query_exception,  "Unknown kv index: ${t} ${i}", ("t", p.table)("i", p.index_name));
+
+   // Is point query of ranged query?
+   bool point_query = p.index_value.has_value() && !p.index_value.value().empty();
+
+   // compose 17bytes prefix
+   vector<char> prefix;
+   prefix.resize( prefix_size() );
+   make_prefix(p.table, p.index_name, 1, prefix);
+
+   abi_serializer abis;
+   abis.set_abi(abi, abi_serializer::create_yield_function(abi_serializer_max_time));
+
+   get_table_rows_result result;
+   uint32_t value_size;
+
+   auto cur_time = fc::time_point::now();
+   auto end_time = cur_time + fc::microseconds(1000 * 10);
+   auto wait_time = end_time - cur_time;
+   ///////////////////////////////////////////////////////////
+   // point query
+   ///////////////////////////////////////////////////////////
+   if( point_query ) {
+      const string &index_value = *p.index_value;
+      vector<char> key;
+      key.resize(prefix.size() + index_value.size());
+      memcpy(key.data(), prefix.data(), prefix.size());
+      memcpy(key.data() + prefix.size(), index_value.data(), index_value.size());
+
+      auto success = kv_context.kv_get(p.code.to_uint64_t(), key.data(), key.size(), value_size);
+      if( success ) {
+         vector<char> value;
+         value.resize(value_size);
+         auto return_size = kv_context.kv_get_data(0, value.data(), value_size);
+         EOS_ASSERT(return_size == value_size, chain::contract_table_query_exception,  "point query value size mismatch: ${s1} ${s2}", ("s1", return_size)("s2", value_size));
+
+         fc::variant row_var;
+         if( p.json ) {
+            try {
+               row_var = abis.binary_to_variant( p.table.to_string(), value, abi_serializer::create_yield_function( wait_time ), shorten_abi_errors );
+            } catch (fc::exception &e) {
+               elog( "Error occurred while converting binary to variant in get_kv_table_rows: '${t}') to: ${i}", ("t", p.table.to_string())("i", p.index_name.to_string()));
+               row_var = fc::variant( value );
+            }
+         } else {
+            row_var = fc::variant( value );
+         }
+         result.rows.emplace_back( std::move(row_var) );
+      }
+      return result;
+   }
+
+   ///////////////////////////////////////////////////////////
+   // ranged query
+   ///////////////////////////////////////////////////////////
```

Comment:
What happens if I specify index_value, lower_bound, and upper_bound? The way I'm reading this, it will do the value lookup as well as the lb/ub range. That doesn't seem like the desired behaviour, so maybe the range query should be in an `else` clause?

Reply:
There is "return result" inside point query.  If point query is successful, it returns.